src/test/java/hillel/spring/service/GreetingServiceTest.java

```diff
@@ -0,0 +1,51 @@
+package hillel.spring.service;
+
+import hillel.spring.repository.GreetingRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.data.Percentage;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {GreetingService.class, GreetingRepository.class})
+public class GreetingServiceTest {
+
+    private GreetingService greetingService;
+
+    @Autowired
+    public void setGreetingService(GreetingService greetingService) {
+        this.greetingService = greetingService;
```

Comment:
It is common to declare fields as public or package private and use `@Autowired` directly on the field (just to avoid creating unnecessary setters in tests)